plugins/quickicon/eos/src/Extension/Eos.php

```diff
@@ -0,0 +1,382 @@
+        } catch (Exception $e) {
+            echo $e->getMessage();
+            exit();
+        }
+        // The message as quickicon
+        // Add the icon to the result array
+        $result               = $event->getArgument('result', []);
+        $messageTextQuickIcon = Text::sprintf($this->currentMessage['quickiconText'], HTMLHelper::_('date', Eos::EOS_DATE, Text::_('DATE_FORMAT_LC3')));
+
+        // The message as quickicon
+
+        $result[] = [
+            [
+                'link'  => $this->currentMessage['messageLink'],
+                'image' => 'fa fa-life-ring',
+                'text'  => $messageTextQuickIcon,
+                'id'    => 'plg_quickicon_eos',
+                'group' => $this->currentMessage['groupText'],
+                'class' => 'success',
+            ],
+        ];
+
+        $event->setArgument('result', $result);
+    }
+
+    /**
+     * Save the plugin parameters
+     *
+     * @return  bool
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    private function saveParams(): bool
+    {
+        $db    = Factory::getContainer()->get('DatabaseDriver');
+        $query = $db->getQuery(true)->update($db->quoteName('#__extensions'))->set($db->quoteName('params') . ' = ' . $db->quote($this->params->toString('JSON')))->where($db->quoteName('type') . ' = ' . $db->quote('plugin'))->where($db->quoteName('folder') . ' = ' . $db->quote('quickicon'))->where($db->quoteName('element') . ' = ' . $db->quote('eos'));
+        try {
+            // Lock the tables to prevent multiple plugin executions causing a race condition
+            $db->lockTable('#__extensions');
+        } catch (Exception $e) {
+            // If we can't lock the tables it's too risky to continue execution
+            return false;
+        }
+        try {
+            // Update the plugin parameters
+            $result = $db->setQuery($query)->execute();
+            $this->clearCacheGroups();
+        } catch (Exception $e) {
+            // If we failed to execute
+            $db->unlockTables();
+            $result = false;
+        }
+        try {
+            // Unlock the tables after writing
+            $db->unlockTables();
+        } catch (Exception $e) {
+            // If we can't lock the tables assume we have somehow failed
+            $result = false;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Determines if the message and quickicon should be displayed
+     *
+     * @return  bool
+     *
+     * @since __DEPLOY_VERSION__
+     *
+     * @throws Exception
+     */
+    private function shouldDisplayMessage(): bool
+    {
+        if (!$this->getApplication()->isClient('administrator') || Factory::getApplication()->getIdentity()->guest || $this->document->getType() !== 'html' || $this->getApplication()->input->getCmd('tmpl', 'index') === 'component' || $this->getApplication()->input->get('option') !== 'com_cpanel') {
+            return false;
+        }
```

Comment:
```suggestion
        if (!$this->getApplication()->isClient('administrator')) {
            return false;
        }

        if ($this->getApplication()->getIdentity()->guest) {
            return false;
        }

        if ($this->document->getType() !== 'html') {
            return false;
        }

        if ($this->getApplication()->getInput()->getCmd('tmpl', 'index') === 'component') {
            return false;
        }

        if ($this->getApplication()->getInput()->get('option') !== 'com_cpanel') {
            return false;
        }
```
use `$this->getApplication()`
use `$this->getApplication()->getInput`; input property is deprecated
code style (line length)

Reply:
@heelc29 

```
if ($this->document->getType() !== 'html') {
    return false;
 }
```
Why not using?
```
if $this->getApplication()->getDocument()->getType() !== 'html') {
    return false;
}
```

Reply:
@sandewt I ask already in the old PR https://github.com/joomla/joomla-cms/pull/40556#discussion_r1186895348

This comment was only about code style 😀